docs/src/main/paradox/http-server.md

```diff
@@ -1,39 +1,55 @@
 
+Notice that we've separated out the `UserRoutes` trait, in which we'll put all our actual route definitions.
+This is a good pattern to follow, especially once your application starts to grow and you'll need some form of 
+compartmentalizing them into groups of routes handling specific parts of the exposed API.
+
+
 ## Binding endpoints
 
 Each Akka HTTP `Route` contains one or more `akka.http.scaladsl.server.Directives`, such as: `path`, `get`, `post`, `complete`, etc. There is also a [low-level API](http://doc.akka.io/docs/akka-http/current/scala/http/low-level-server-side-api.html) that allows to inspect requests and create responses manually. For the user registry service, the example needs to support the actions listed below. For each, we can identify a path, the HTTP method, and return value:
 
-|Functionality       | Path       | HTTP method     | Returns              |
-|--------------------|------------|-----------------|----------------------|
-| Create a user      | /users     | POST            | Confirmation message |
-| Retrieve a user    | /users/$ID | GET             | JSON payload         |
-| Remove a user      | /users/$ID | DELETE          | Confirmation message |
-| Retrieve all users | /users     | GET             | JSON payload         |
+|Functionality       | HTTP Method | Path       | Returns              |
+|--------------------|-------------|------------|----------------------|
+| Create a user      | POST        | /users     | Confirmation message |
+| Retrieve a user    | GET         | /users/$ID | JSON payload         |
+| Remove a user      | DELETE      | /users/$ID | Confirmation message |
+| Retrieve all users | GET         | /users     | JSON payload         |
 
-In the `QuickstartServer` source file, the definition of the `Route` begins with the line:
-`lazy val routes: Route =`.
+In the `QuickstartServer` source file, the definition of the `Route` delegates to the routes defined in `UserRoutes`:
+`lazy val routes: Route = userRoutes`.
+
+In larger applications this is where we'd combine the various routes of our application into a big route that is concatenating
+the various routes of our services. We'd do this using the concat directive like this: `val route = concat(userRoutes, healthCheckRoutes, ...)`
 
 Let's look at the pieces of the example `Route` that bind the endpoints, HTTP methods, and message or payload for each action.
 
 ### Retrieving and creating users
 
 The definition of the endpoint to retrieve and create users look like the following:
 
-@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #users-get-post }
+@@snip [UserRoutes.scala]($g8src$/scala/com/lightbend/akka/http/sample/UserRoutes.scala) { #users-get-post }
 
+A Route is constructed by nesting various *directives* which route an incoming request to the apropriate handler block.
 Note the following building blocks from the snippet:
 
 **Generic functionality**
```

Comment:
You just mentioned "directives", but it's not mention that `pathPrefix`, `pathEnd` are directives.